consul/util.go

```diff
@@ -153,10 +153,37 @@ func isPrivateIP(ip_str string) bool {
 	return false
 }
 
+// Returns addresses from interfaces that is up and supports
+// both broadcast and multicast
+func activeInterfaceAddresses() ([]net.Addr, error) {
+	var out []net.Addr
+
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get interfaces: %v", err)
+	}
+
+	reqFlags := net.FlagUp | net.FlagBroadcast | net.FlagMulticast
```

Comment:
Reasoning for filtering on broadcast and multicast was to filter out loopback and other dummy interfaces, could change that to be a explicit filter on `FlagLoopback` instead.

Windows support all these flags, see https://golang.org/src/net/interface_windows.go.